bin/start-systemtest

```diff
@@ -0,0 +1,206 @@
+  local DATASET=$(get_dataset "${TASK}")
+
+  echo "($(get_timestamp)) [${TEST_VERSION}] Checking task situation [${ID}]..."
+
+  if [[ ${TEST_VERSION} == 'endtoend-test' && ${DATASET} == 'landsat_7' ]]; then
+    echo "($(get_timestamp)) [${TEST_VERSION}] Test failure, but this is expected."
+  elif [[ ${STATE} == 'archived' ]]; then
+    echo "($(get_timestamp)) [${TEST_VERSION}] Successful test."
+    check_files ${ID} ${TEST_VERSION}
+  else
+    echo "($(get_timestamp)) [${TEST_VERSION}] Test failure."
+    exit 1
+  fi
+}
+
+check_files(){
+  local ID=${1}
+  local TEST_VERSION=${2}
+
+  if [[ ${TEST_VERSION} == 'endtoend-test' ]]; then
+    check_md5sum_files ${ID} ${TEST_VERSION}
+  else
+    echo "($(get_timestamp)) [${TEST_VERSION}] The ${TEST_VERSION} version does not need to check the files."
+  fi
+}
+
+check_tasks(){
+  local TASK_IDS=${1}
+  local TEST_VERSION=${2}
+  local TIMESLEEP=$(get_timesleep ${TEST_VERSION})
+
+  echo "($(get_timestamp)) [${TEST_VERSION}] Checking submitted tasks..."
+
+  for ID in $(echo "${TASK_IDS}" | jq ".[]" | tr -d \")
+  do
+    while true; do
+       local TASK=$(get_task_by_id ${ID})
+       local STATE=$(get_state "${TASK}")
+
+       echo "($(get_timestamp)) [${TEST_VERSION}] Current task state [${ID}]: ${STATE}"       
+       if [ ${STATE} == 'archived' -o ${STATE} == 'failed' ]; then
+         echo "($(get_timestamp)) [${TEST_VERSION}] Task [${ID}] has finished processing."
+         check_situation ${ID} ${TEST_VERSION}
+         break
+       fi
+       echo "($(get_timestamp)) [${TEST_VERSION}] Waiting ${TIMESLEEP} seconds to check again..."
+       sleep ${TIMESLEEP}
+    done
+  done
+}
+
+submit_task(){
+  local TEST_VERSION=${1}
+  
+  bash ${SUBMIT_TASK_SCRIPT} ${USER_EMAIL} ${USER_PASSWORD} -7.413 -7.047 -37.314 -36.257 2015-06-23 2015-06-23 ${TEST_VERSION} ${TEST_VERSION} ${TEST_VERSION} ${DISPATCHER_ADDRESS}
+}
+
+get_links(){
+  local ID=${1}
+
+  bash ${GET_TASK_LINKS_SCRIPT} ${USER_EMAIL} ${USER_PASSWORD} ${ID} ${DISPATCHER_ADDRESS}
+}
+
+get_name(){
+  local LINK=${1}
+  local NAME=$(echo ${LINK} | cut -d "*" -f1)
+
+  echo "${NAME}"
+}
+
+get_url(){
+  local LINK=${1}
+  local URL=$(echo ${LINK} | cut -d "*" -f2)
+
+  echo "${URL}"
+}
+
+get_md5sum(){
+  local ULR=${1}
+  MD5SUM=$(curl --silent "${URL}" | md5sum | awk '{print $1}')
+
+  echo "${MD5SUM}"
+}
+
+check_md5sum_files(){
+  local ID=${1}
+  local TEST_VERSION=${2}
+
+  echo "($(get_timestamp)) [${TEST_VERSION}] Checking md5sum of task files [${ID}]..."
+  LINKS=$(get_links ${ID})
+
+  for NAME_URL in $(echo $LINKS | jq '[ .[] | (.name + "*" + .url) ]' | jq '.[]' | tr -d \"); do
+    NAME=$(get_name $NAME_URL)
+    URL=$(get_url $NAME_URL)
+    FILENAME=$(basename $NAME)
+
+    RESULT=$(grep ${FILENAME} ${SYSTEMTEST_MD5SUM_RESULTS_FILE})
+    EXIT_CODE=$?
+  
+    if [[ "$EXIT_CODE" == "0" ]]; then
+      MD5SUM=$(curl --silent "$URL" | md5sum | awk '{print $1}')
+      MD5SUM=${MD5SUM##*()}
```

Comment:
add a comment explaining this parsing as well. It took a lot of time of us to figure out that it might be necessary

Reply:
done